src/components/Sidebar/BranchTransition.tsx

```diff
@@ -1,21 +1,119 @@
-import React, { FC, memo } from 'react';
+import React, { FC, memo, useState, useCallback, ChangeEvent, useEffect, useRef } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPlus, faTools } from '@fortawesome/free-solid-svg-icons';
-
+import DropDown from 'components/elements/DropDown';
+import { TextField, FormControl } from '@material-ui/core';
+import { setTransitionCriteria, setTransitionCriteriaDisplay } from 'utils/builder';
 import { SidebarHeader, SidebarButton } from '.';
 import { Pathway, Transition } from 'pathways-model';
 import useStyles from './styles';
 
 interface BranchTransitionProps {
   pathway: Pathway;
+  currentNodeKey: string;
   transition: Transition;
   updatePathway: (pathway: Pathway) => void;
 }
 
-const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, updatePathway }) => {
+const criteriaOptions = [
+  { value: 'TumorSize2+', label: 'Tumor Size > 2cm' },
+  { value: 'TumorSize2-', label: 'Tumor Size <= 2cm' },
+  { value: 'Other', label: 'Other' }
+];
+
+function usePreviousString(value: string): string {
+  const ref = useRef<string>();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current || '';
+}
+
+function usePreviousPathway(value: Pathway): Pathway {
+  const ref = useRef<Pathway>();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current || value;
+}
+
+const BranchTransition: FC<BranchTransitionProps> = ({
+  pathway,
+  currentNodeKey,
+  transition,
+  updatePathway
+}) => {
   const styles = useStyles();
   const transitionKey = transition?.transition || '';
   const transitionNode = pathway.states[transitionKey];
+  const [useSelected, setUseSelected] = useState<boolean>(false);
```

Comment:
Can we rename this variable to be a little more obvious what it does. It looks like it is used to represented whether or not use criteria button has been selected.